050-health-endpoint/proposal.md

```diff
@@ -0,0 +1,63 @@
+# Summary
+
+A public HTTP endpoint that gives basic information about the health of Concourse
+ cluster.
+
+
+# Motivation
+
+There are some scenarios in which a Concourse cluster is part of bigger setup for
+automation. In such scenario other parts of the system rely on Concourse not
+only for CI/CD, but for other type of automations as well. In order to have the
+whole process working the system also monitors the availability of its parts and
+ takes measures if some of its parts are not operational (for example executes
+some predefined steps - sends informing mails, triggers alerts, executes
+medication steps, etc). In such cases it is important for the system to be able
+to determine the state of every part of it. So it would be nice if the parts of
+this system have a common way to return their health/availability status.
+
+Currently there is no easy way for external (monitoring) system to understand
```

Comment:
Adding some more context to this for those outside the team

the way that we (the Concourse team itself) has been dealing with the *"is this installation good or not?"* question has been mostly through the use of the metrics that we expose, and SLIs (https://github.com/concourse/oxygen-mask - which relies on another Concourse installation sending probes to the first - / https://github.com/cirocosta/slirunner - doesn't require another installation, but is k8s and prometheus-first) which takes the approach of verifying the healthiness by sending specific workloads to it that would attest in an end-to-end fashion whether it's properly working or not.

what I like about this kind of approach is that you can keep that external to the main system, and thus, build the integrations to it however you want (e.g., in the case of `slirunner`, it exposes the information through Prometheus, but if you'd prefer to have an endpoint being hit when things fail, you could build that too, all while still leveraging "the core concourse" under the hood), as well as highlighting which kinds of user workflows are currently broken / degredated 🤔